### PR TITLE
LSP: Completions for Erlang module exports (BT-680)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Implements LSP completions for `Erlang <module>` expressions, completing Phase 1 of ADR 0028.

**Linear issue:** https://linear.app/beamtalk/issue/BT-680

### New: `erlang_modules.rs` — Static export data
- Curated exports for 15 common OTP modules: `lists`, `maps`, `erlang`, `string`, `io`, `io_lib`, `math`, `timer`, `file`, `ets`, `gen_server`, `proplists`, `rand`, `os`, `binary`
- Arity-to-selector conversion: `reverse/1` → `reverse:`, `seq/2` → `seq:with:`
- Module descriptions and function detail strings

### Modified: `completion_provider.rs` — Erlang context detection
- Detects `Erlang ` at cursor → suggests module names (CompletionKind::Module)
- Detects `Erlang <module> ` at cursor → suggests exports as Beamtalk selectors
- Text-based detection with AST fallback for robustness
- Graceful fallback for unknown modules (no completions, no error)

### Tests
- 19 new unit tests covering module suggestions, export suggestions, deduplication, detail/documentation, unknown modules, assignment context, and context detection helpers
- 7 unit tests for selector conversion and module lookup

## How it works

```
Erlang |        → suggests: lists, maps, erlang, string, io, math, ...
Erlang lists |  → suggests: reverse:, map:with:, foldl:with:with:, sort:, ...
Erlang unknown | → falls back to normal completions
```

## Summary by CodeRabbit

* **New Features**
  * Context-aware Erlang completions: detects Erlang contexts and suggests module names and exports with Erlang-style signatures and details.
  * Prepopulated common OTP modules for faster suggestions; deduplicates overloaded selectors and works in assignment and class-reference contexts.

* **Tests**
  * Added comprehensive tests covering detection, module suggestions, export listings, documentation, deduplication, and edge cases.

### CODE REVIEW

File: crates/beamtalk-core/src/queries/erlang_modules.rs [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+//! Since the LSP has no runtime connection to query `module_info(exports)`,
+//! we maintain a curated list of commonly-used OTP module exports.
+//!
+//! ## Selector Conversion
+//!
+//! Erlang function/arity pairs are converted to Beamtalk keyword selectors:
+//! - `reverse/1` → `reverse:`
+//! - `seq/2` → `seq:with:`
+//! - `seq/3` → `seq:with:with:`
+//! - `node/0` → `node` (unary)
```
[the remark this answers is not in the document]
Comment (jamesc) [reply crates/beamtalk-core/src/queries/erlang_modules.rs]: Fixed in cb9ca75. Changed 'unary' to 'nullary' for the 0-arity example.
